python-checks/src/main/java/org/sonar/python/checks/RandomSeedCheck.java

```diff
@@ -67,14 +74,47 @@ public void initialize(Context context) {
 
   private void checkEmptySeedCall(SubscriptionContext ctx) {
     CallExpression call = (CallExpression) ctx.syntaxNode();
-    Optional.ofNullable(call.calleeSymbol()).map(Symbol::fullyQualifiedName)
-        .map(SEED_METHODS_TO_CHECK::get)
-        .filter(argName -> isSeedArgumentAbsentOrNone(argName, call.arguments()))
-        .ifPresent(fqn -> ctx.addIssue(call, MESSAGE));
+
+    Optional<Symbol> maybeCalleeSymbol = Optional.ofNullable(call.calleeSymbol());
+
+    maybeCalleeSymbol
+      .map(Symbol::fullyQualifiedName)
+      .map(SEED_METHODS_TO_CHECK::get)
+      .filter(argName -> isArgumentAbsentOrNone(TreeUtils.nthArgumentOrKeyword(0, argName, call.arguments())))
+      .map(arg -> MESSAGE)
+      .or(() -> maybeCalleeSymbol
+        .filter(symbol -> symbol.fullyQualifiedName() != null && symbol.fullyQualifiedName().startsWith(SKLEARN_FQN))
+        .filter(RandomSeedCheck::hasRandomStateParameter)
+        .filter(symbol -> isArgumentAbsentOrNone(TreeUtils.argumentByKeyword(SKLEARN_ARG_NAME, call.arguments())))
+        .map(symbol -> SKLEARN_MESSAGE))
+      .ifPresent(message -> ctx.addIssue(call, message));
+  }
+
+  private static boolean hasRandomStateParameter(Symbol calleeSymbol) {
+    if (calleeSymbol.is(Kind.CLASS)) {
+      List<Symbol> constructorSymbols = ((ClassSymbolImpl) calleeSymbol).declaredMembers()
+        .stream()
+        .filter(s -> "__init__".equals(s.name()))
+        .toList();
+      if (constructorSymbols.size() == 1) {
+        return hasRandomStateParameter(constructorSymbols.get(0));
+      }
+      return false;
+    }
+    if (calleeSymbol.is(Kind.FUNCTION)) {
+      List<FunctionSymbol> symbols = SymbolUtils.getFunctionSymbols(calleeSymbol);
```

Comment:
Is there a reason not to just cast the current symbol ?

Reply:
No not really I think using the optional API will in fact make things clearer as well